### PR TITLE
ktx: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

File: recipes/ktx/all/patches/0004-unvendor-dependencies.patch
Comment (RubenRBS) [inline recipes/ktx/all/patches/0004-unvendor-dependencies.patch:1]: I see this patch is new. This the dependencies just not work in the past?

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot!

File: recipes/ktx/all/patches/0004-unvendor-dependencies.patch
Comment (SpaceIm) [reply recipes/ktx/all/patches/0004-unvendor-dependencies.patch:1]: It worked but it was relying on `link_libraries(${CONAN_LIBS})` in CMakeLists wrapper, which is specific to legacy `cmake` generator, so it must be addressed differently when you use `CMakeToolchain`. I've also moved some replace_in_file to this patch.

Comment (RubenRBS) [reply recipes/ktx/all/patches/0004-unvendor-dependencies.patch:1]: Nice! Thanks for the explanation

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**